server/channels/app/platform/web_hub.go

```diff
@@ -637,7 +637,22 @@ func (i *hubConnectionIndex) Has(wc *WebConn) bool {
 
 // ForUser returns all connections for a user ID.
 func (i *hubConnectionIndex) ForUser(id string) []*WebConn {
-	return i.byUserId[id]
+	// Fast path if there is only one or fewer connection.
+	if len(i.byUserId[id]) <= 1 {
+		return i.byUserId[id]
+	}
+	// If there are multiple connections per user,
+	// then we have to return a clone of the slice
+	// to allow connIndex.Remove to be safely called while
+	// iterating the slice.
```

Comment:
Not sure if this is worth optimizing further but we could technically queue up the removals and apply them once `broadcast()` returns to avoid the clone. Just a thought for the future.

Reply:
I had thought of that. Didn't like it, and abandoned it. :)